v2/tracing_test.go

```diff
@@ -68,6 +70,15 @@ func Test_NewTracingHandler(t *testing.T) {
 			hasTraceContextOnContext: true,
 			hasTraceContextOnMessage: true,
 		},
+		{
+			name:                     "has traceCtx on context - custom extractFn",
+			hasTraceContextOnContext: true,
+			hasTraceContextOnMessage: false,
+			extractFn: func(ctx context.Context, message *azservicebus.ReceivedMessage) (context.Context, trace.Span) {
+				ctx, span := gootel.Tracer("test-tracer").Start(ctx, "test-span")
```

Comment:
your example is taking the trace from the receiver context.
but not extracting the context fron the message anymore, so you lose the trace propagation from sender to receiver.